math/src/main/kotlin/ktx/math/ImmutableVector.kt

```diff
@@ -0,0 +1,181 @@
+    fun mulAdd(v: T, mulVec: T): T = this + (v * mulVec)
+
+    @Deprecated(MUTABLE_METHOD_DEPRECATION_MESSAGE, ReplaceWith("withLimit(limit)"), DeprecationLevel.ERROR)
+    fun limit(limit: Float): T = withLimit(limit)
+
+    @Deprecated(MUTABLE_METHOD_DEPRECATION_MESSAGE, ReplaceWith("withClamp(min, max)"), DeprecationLevel.ERROR)
+    fun clamp(min: Float, max: Float): T = withClamp(min, max)
+
+    @Deprecated(MUTABLE_METHOD_DEPRECATION_MESSAGE, ReplaceWith("v"), DeprecationLevel.ERROR)
+    fun set(v: T): T = v
+
+    @Deprecated(MUTABLE_METHOD_DEPRECATION_MESSAGE, ReplaceWith("withInterpolation(target, alpha, interpolator)"), DeprecationLevel.ERROR)
+    fun interpolate(target: T, alpha: Float, interpolator: Interpolation): T = withInterpolation(target, alpha, interpolator)
+
+    @Deprecated(MUTABLE_METHOD_DEPRECATION_MESSAGE, ReplaceWith("withLength(len)"), DeprecationLevel.ERROR)
+    fun setLength(len: Float): T = withLength(len)
+
+    @Deprecated(MUTABLE_METHOD_DEPRECATION_MESSAGE, ReplaceWith("withLimit2(limit2)"), DeprecationLevel.ERROR)
+    fun limit2(limit2: Float): T = withLimit2(limit2)
+
+    @Deprecated(MUTABLE_METHOD_DEPRECATION_MESSAGE, ReplaceWith("this - v"), DeprecationLevel.ERROR)
+    fun sub(v: T): T = this - v
+
+    @Deprecated(MUTABLE_METHOD_DEPRECATION_MESSAGE, ReplaceWith("nor"), DeprecationLevel.ERROR)
+    fun nor(): T = nor
+}
+
+/** Returns the distance between this and the [other] vector */
+inline infix fun <T : ImmutableVector<T>> T.dst(other: T): Float = sqrt(dst2(other))
+
+/** Returns this vector scaled by (1 / [scalar]) */
+inline operator fun <T : ImmutableVector<T>> T.div(scalar: Float): T = times(1 / scalar)
+
+/** Returns true if this vector is collinear with the [other] vector */
+fun <T : ImmutableVector<T>> T.isCollinear(other: T, epsilon: Float = MathUtils.FLOAT_ROUNDING_ERROR): Boolean =
+        isOnLine(other, epsilon) && hasSameDirection(other)
+
+/** Returns true if this vector is opposite collinear with the [other] vector */
+fun <T : ImmutableVector<T>> T.isCollinearOpposite(other: T, epsilon: Float = MathUtils.FLOAT_ROUNDING_ERROR): Boolean =
+        isOnLine(other, epsilon) && hasOppositeDirection(other)
+
+/** Returns true if this vector is opposite perpendicular with the [other] vector */
+fun <T : ImmutableVector<T>> T.isPerpendicular(other: T, epsilon: Float = MathUtils.FLOAT_ROUNDING_ERROR): Boolean =
+        MathUtils.isZero(dot(other), epsilon)
+
+/** Returns whether this vector has similar direction compared to the other vector. */
+fun <T : ImmutableVector<T>> T.hasSameDirection(other: T): Boolean =
+        dot(other) > 0f
+
+/** Returns whether this vector has opposite direction compared to the other vector. */
+fun <T : ImmutableVector<T>> T.hasOppositeDirection(other: T): Boolean =
+        dot(other) < 0f
+
+internal const val MUTABLE_METHOD_DEPRECATION_MESSAGE = "Unlike its equivalent in LibGDX, this function doesn't mutate the vector"
```

Comment:
"Mutate" might not be obvious for programming beginners. I'd go with something along the lines of:

`Unlike its equivalent in LibGDX, this function does not change the internal state of the vector and returns a new instance instead. This might break existing code designed with mutable vectors in mind.`